### PR TITLE
Remove value constructors for ConsoleValueRef & fix callbacks

### PR DESCRIPTION
**Pull request by jamesu**
Fixes issue #1289

### CODE REVIEW

File: Engine/source/console/simEvents.cpp [hunk 1]
```diff
@@ -129,7 +129,7 @@ ConsoleValueRef SimConsoleThreadExecCallback::waitForResult()
       return retVal;
    }
 
-   return (const char*)NULL;
+   return ConsoleValueRef();
```
Comment (eightyeight) [inline Engine/source/console/simEvents.cpp]: As far as I can tell, this should never be called, right?

**Comment by eightyeight** [comment]
Will merge this manually after review.

File: Engine/source/console/simEvents.cpp (hunk 1 above)
Comment (jamesu) [inline Engine/source/console/simEvents.cpp]: I'm guessing no but it doesn't make sense to return retVal if for whatever reason acquire fails.

Comment (eightyeight) [inline Engine/source/console/simEvents.cpp]: Yep, I was just trying to understand the flow.

**Comment by eightyeight** [comment]
Yay it works. Thanks @jamesu! Merged in 12d435c.

**Pull request closed without merging**